src/worker.cpp

```diff
@@ -968,6 +1013,15 @@ types::VPDMapVariant Worker::parseVpdFile(const std::string& i_vpdFilePath)
             }
         }
 
+        types::ObjectMap l_objectInterfaceMap;
+        primeInventory(i_vpdFilePath, l_objectInterfaceMap);
```

Comment:
Priming of inventory should always happen in case there is any failure in collection of the FRU.
In current situation, in case there is a throw from line 1010. I don't see "primeInventory" getting executed in that case. Please find another appropriate place to call your API.

Reply:
Both PrimeInventory and executePostFailAction should be called if i_vpdFilePath file doesn't exists.
even if I call primeInventory first and later executePostFailAction, it will impact executePostFailAction if there is any exception from primeInventory.
So another place we can call primeInventory is in parseAndPublishVPD catch block. 
But am not sure, is it correct to do so. @SunnySrivastava1984  please provide me your input here.

Reply:
> But am not sure

What are your doubts?

Reply:
Shall I call PrimeInventory from  parseAndPublishVPD's catch block ?

Reply:
That is the answer you need to come up with?

I just suggested a place from where you can call but at the end of the day, requirement is,
- In case there is any failure in the process of collecting a FRU then priming needs to be done for that FRU. 

So if you think that is the best place, go ahead, if not then see what can be other place(s) from where it can be called.